aws unseal method raises error "panic: errors: *target must be interface or implement error"

**Describe the bug**:
I am trying to bring up vault using aws kms and s3 automatic unseal.
i have deployed the vault-operator helm chart, created an aws role with permissions to access the kms and s3, enabled the IAM OIDC provider on the eks cluster, and gave the vault service account the role to assume (all as stated here: https://banzaicloud.com/docs/bank-vaults/cloud-permissions/)
after that, when deploying vault object with "aws" as unseal mechanism, i get the following error from the "bank-vaults" container inside the vault pod, and no seal keys/root tokens are being created in the s3 bucket:
```
time="2021-05-05T13:23:57Z" level=info msg="initializing vault..."
time="2021-05-05T13:23:57Z" level=info msg="vault metrics exporter enabled: :9091/metrics"
time="2021-05-05T13:23:57Z" level=info msg="initializing vault"
panic: errors: *target must be interface or implement error

goroutine 1 [running]:
errors.As(0x239a080, 0xc0007121b0, 0x1dbeda0, 0xc0007245a0, 0xc0004893e0)
	/usr/local/go/src/errors/wrap.go:87 +0x533
emperror.dev/errors.As(...)
	/go/pkg/mod/emperror.dev/errors@v0.8.0/wrap_go1_13.go:26
github.com/banzaicloud/bank-vaults/pkg/kv.IsNotFoundError(0x239a080, 0xc0007121b0, 0xa)
	/build/pkg/kv/kv.go:43 +0x5b
github.com/banzaicloud/bank-vaults/pkg/kv/multi.(*multi).Get(0xc0002a9db8, 0x2077fc0, 0xa, 0x966785, 0xc000195d90, 0x202c5e0, 0xc00047dd50, 0x1)
	/build/pkg/kv/multi/multi.go:52 +0x119
github.com/banzaicloud/bank-vaults/internal/vault.kvTester.Test(0x7fc53f8e7cd8, 0xc0002a9db8, 0x2077fc0, 0xa, 0x1, 0xc0000b3b00)
	/build/internal/vault/operator_client.go:110 +0x49
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).Init(0xc0000b3a70, 0x4, 0xc000515c80)
	/build/internal/vault/operator_client.go:252 +0x13fb
main.glob..func4(0x3158f60, 0xc0004165a0, 0x1, 0x11)
	/build/cmd/bank-vaults/unseal.go:121 +0x462
github.com/spf13/cobra.(*Command).execute(0x3158f60, 0xc000416480, 0x11, 0x12, 0x3158f60, 0xc000416480)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x3158cc0, 0x216b810, 0xc00030a900, 0x3)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.execute()
	/build/cmd/bank-vaults/main.go:133 +0xe7
main.main()
	/build/cmd/bank-vaults/main.go:269 +0x75
```

**Expected behaviour**:
bank-vaults succesfully creates the necessary unseal keys and root token in the s3 bucket

**Steps to reproduce the bug**:
- Created KMS key to encrypt/decrypt S3 bucket
- Created S3 bucket encrypted with the KMS key created prior
- Created IAM Role according to all steps here: https://banzaicloud.com/docs/bank-vaults/cloud-permissions/ (changing values as necessary to fit my environment)
- Deployed vault-operator chart:  
```
helm upgrade --install --namespace vault vault-operator banzaicloud-stable/vault-operator
```

- deployed rbac.yaml in vault namespace:
```
kind: ServiceAccount
apiVersion: v1
metadata:
  name: vault
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::<my_account_number>:role/<vault_iam_role_name>

---

kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vault
rules:
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["*"]
  - apiGroups: [""]
    resources: ["pods"]
    verbs: ["get", "update", "patch"]
---

kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vault
roleRef:
  kind: Role
  name: vault
  apiGroup: rbac.authorization.k8s.io
subjects:
  - kind: ServiceAccount
    name: vault

---

# This binding allows the deployed Vault instance to authenticate clients
# through Kubernetes ServiceAccounts (if configured so).
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: vault-auth-delegator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:auth-delegator
subjects:
  - kind: ServiceAccount
    name: vault
    namespace: vault
```
- deployed cr-aws.yaml in vault namespace:
```
apiVersion: "vault.banzaicloud.com/v1alpha1"
kind: "Vault"
metadata:
  name: "vault"
spec:
  size: 1
  image: vault:1.6.2

  # Insted of credentialsConfig one can use IAM instance profiles, or kube2iam for example:
  annotations:
    iam.amazonaws.com/role: "arn:aws:iam::<my_account_number>:role/<vault_iam_role_name>"

  # Describe where you would like to store the Vault unseal keys and root token
  # in S3 encrypted with KMS.
  unsealConfig:
    aws:
      kmsKeyId: "<kms_key>"
      kmsRegion: "us-east-2"
      s3Bucket: "<s3_bucket_name>"
      s3Prefix: "vault-operator/"
      s3Region: "us-east-2"

  # Specify the ServiceAccount where the Vault Pod and the Bank-Vaults configurer/unsealer is running
  serviceAccount: vault

  # A YAML representation of a final vault config file, this config represents
  # a backend config in AWS.
  # See https://www.vaultproject.io/docs/configuration/ for more information.
  config:
    storage:
      s3:
        region: "us-east-2"
        bucket: "<bucket_name>"
    listener:
      tcp:
        address: "0.0.0.0:8200"
        tls_cert_file: /vault/tls/server.crt
        tls_key_file: /vault/tls/server.key
    api_addr: https://vault.default:8200
    telemetry:
      statsd_address: localhost:9125
    ui: true

  # See: https://banzaicloud.com/docs/bank-vaults/cli-tool/#example-external-vault-configuration
  # The repository also contains a lot examples in the deploy/ and operator/deploy directories.
  externalConfig:
    policies:
      - name: allow_secrets
        rules: path "secret/*" {
          capabilities = ["create", "read", "update", "delete", "list"]
          }
    auth:
      - type: kubernetes
        roles:
          # Allow every pod in the default namespace to use the secret kv store
          - name: default
            bound_service_account_names: default
            bound_service_account_namespaces: default
            policies: allow_secrets
            ttl: 1h

  # If you are not using IAM instance profiles to authenticate against AWS you can pass in an
  # AWS credentials refering to a Secret with the following configuration block. You have to
  # create the "aws-cred" secret manually, it has to have a key called "credentials" holding
  # the configuration in the following format:
  # https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#the-shared-credentials-file
  # credentialsConfig:
  #   env: AWS_SHARED_CREDENTIALS_FILE
  #   path: /etc/aws/credentials
  #   secretName: aws-cred

  # Or you can follow: https://banzaicloud.com/docs/bank-vaults/cloud-permissions/#aws
  # to give access to AWS S3 and KMS on an EKS cluster (recommended).
```

**Additional context**:
i assume i am doing something wrong as i did not find any documentation on this kind of error but after struggling with this for 2 days i can't quite tell.. tried recreating kms/bucket/roles from scratch but this error keeps repeating. would like to know if it is an actual bug or not. thanks!

**Environment details**:
- Kubernetes version (e.g. v1.10.2): v1.19.8
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): AWS
- bank-vaults version (e.g. 0.4.17): vault-operator version: vault-operator-1.12.1
- Install method (e.g. helm or static manifests): vault-operator installed with helm 3, vault with static manifest taken from here: https://github.com/banzaicloud/bank-vaults/blob/master/operator/deploy/cr-aws.yaml
- Logs from the misbehaving component (and any other relevant logs): stated above
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data: stated above

/kind bug

I believe I'm running into the same issue.  Using 1.9.0, I can deploy my CR without issue.  However, if I upgrade to 1.12.0, I'm seeing the bank-vaults container in the Vault pod panic:

```time="2021-05-06T00:45:29Z" level=info msg="initializing vault"
panic: errors: *target must be interface or implement error

goroutine 1 [running]:
errors.As(0x239a080, 0xc0008b0180, 0x1dbeda0, 0xc0008bc430, 0xc00042d6d0)
	/usr/local/go/src/errors/wrap.go:87 +0x533
emperror.dev/errors.As(...)
	/go/pkg/mod/emperror.dev/errors@v0.8.0/wrap_go1_13.go:26
github.com/banzaicloud/bank-vaults/pkg/kv.IsNotFoundError(0x239a080, 0xc0008b0180, 0xa)
	/build/pkg/kv/kv.go:43 +0x5b
github.com/banzaicloud/bank-vaults/pkg/kv/multi.(*multi).Get(0xc00000e690, 0x2077fc0, 0xa, 0x966785, 0xc000197d90, 0x202c5e0, 0xc0000bfea0, 0x1)
	/build/pkg/kv/multi/multi.go:52 +0x119
github.com/banzaicloud/bank-vaults/internal/vault.kvTester.Test(0x7f8b1ad722f8, 0xc00000e690, 0x2077fc0, 0xa, 0x1, 0xc00038dbc0)
	/build/internal/vault/operator_client.go:110 +0x49
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).Init(0xc00038db30, 0x4, 0xc0004ffc80)
	/build/internal/vault/operator_client.go:252 +0x13fb
main.glob..func4(0x3158f60, 0xc00021b0e0, 0x1, 0x11)
	/build/cmd/bank-vaults/unseal.go:121 +0x462
github.com/spf13/cobra.(*Command).execute(0x3158f60, 0xc00021afc0, 0x11, 0x12, 0x3158f60, 0xc00021afc0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x3158cc0, 0x216b810, 0xc00030cb40, 0x3)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.execute()
	/build/cmd/bank-vaults/main.go:133 +0xe7
main.main()
	/build/cmd/bank-vaults/main.go:269 +0x75
```

--- a/pkg/kv/kv.go
+++ b/pkg/kv/kv.go
@@ -20,26 +20,26 @@ import (
 	"emperror.dev/errors"
 )
 
-// NotFoundError represents an error when a key is not found
-type NotFoundError struct {
+// notFoundError represents an error when a key is not found
+type notFoundError struct {
 	msg string // description of error
 }
 
-func (*NotFoundError) NotFound() bool {
+func (notFoundError) NotFound() bool {
 	return true
 }
 
-func (e *NotFoundError) Error() string { return e.msg }
+func (e notFoundError) Error() string { return e.msg }
 
 // NewNotFoundError creates a new NotFoundError
-func NewNotFoundError(msg string, args ...interface{}) *NotFoundError {
-	return &NotFoundError{
+func NewNotFoundError(msg string, args ...interface{}) error {
+	return notFoundError{
 		msg: fmt.Sprintf(msg, args...),
 	}
 }
 
 func IsNotFoundError(err error) bool {
-	var notFoundErr NotFoundError
+	var notFoundErr notFoundError
 	if errors.As(err, &notFoundErr) && notFoundErr.NotFound() {
 		return true
 	}

--- a/pkg/kv/kv_test.go
+++ b/pkg/kv/kv_test.go
@@ -1,0 +1,29 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsNotFoundError(t *testing.T) {
+	nerr := NewNotFoundError("I'm not found")
+	assert.True(t, IsNotFoundError(nerr))
+
+	assert.False(t, IsNotFoundError(io.EOF))
+}

